core/modules/user/user.module

```diff
@@ -3086,6 +2986,25 @@ function user_register_submit($form, &$form_state) {
 }
 
 /**
+ * Implements hook_user_insert().
+ */
+function user_user_insert($account) {
+  // Don't create a new username if one is already set.
+  if (!empty($account->name) && strpos($account->name, 'account-') !== 0) {
```

Comment:
Good point! Maybe we should put something longer and more likely to be unique in as a placeholder...